### PR TITLE
Uniform line endings for multi line strings/comments

### LINKED ISSUE #6501: CRLF: CR is removed in multiline comments and strings
```
type IAmIncredibleLongParameterType = {};
const IAmAnotherFunctionName = (_0: IAmIncredibleLongParameterType) => {};
export const IAmIncredibleLongFunctionName = IAmAnotherFunctionName(
    (_0: IAmIncredibleLongParameterType) => {
        setTimeout(() => {
            /*
                Multiline comment
                Multiline comment
                Multiline comment
            */
            console.log(
                'Multiline string\
                 Multiline string\
                 Multiline string'
            );
        });
    }
);
```
**Before running `lint --fix`:**

before

Errors:
```
   6:15  error  Delete `␍`  prettier/prettier
   7:34  error  Delete `␍`  prettier/prettier
   8:34  error  Delete `␍`  prettier/prettier
   9:34  error  Delete `␍`  prettier/prettier
  12:35  error  Delete `␍`  prettier/prettier
  13:35  error  Delete `␍`  prettier/prettier
```
**After running `lint --fix`:**

after

**What version of `eslint` are you using?**
6.0.1

**What version of `prettier` are you using?**
1.18.2

**What version of `eslint-plugin-prettier` are you using?**
3.1.0

**Please paste any applicable config files that you're using (e.g. `.prettierrc` or `.eslintrc` files)**
```
{
  "printWidth": 100,
  "tabWidth": 4,
  "singleQuote": true,
  "quoteProps": "consistent"
}
```
```
module.exports = {
    root: true,
    parser: '@typescript-eslint/parser',
    parserOptions: {
        ecmaVersion: 6,
        sourceType: 'module'
    },
    plugins: ['import', 'folder-schema', 'processors-stub', 'react-hooks'],
    extends: ['plugin:import/typescript', 'plugin:prettier/recommended'],
    rules: {
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn'
    }
};
```

**What source code are you linting?**
*.tsx

**What did you expect to happen?**
Equal line endings for the whole file.

**What actually happened?**
The plugin is requiring and replacing CRLF with LF in multiline strings and comments

### PR DESCRIPTION
**Pull request by sidharthv96**
Fixes #6501 
* Removed double parsing of the doc
* Moved newline check to printDocToString

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/end_of_line/multiline.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const foo = bar(
+  (_0: baz) => {
```
Comment (thorn0) [inline tests/end_of_line/multiline.js]: I can't reproduce the bug in 2.0.2 with this snippet. Please use the original snippet from #6501 as a test instead.

Comment (sidharthv96) [reply tests/end_of_line/multiline.js]: Done.

**Comment by j-f1** [comment]
How does this work with e.g. template strings containing LFs? Does it preserve them or does it change their content?

**Comment by sidharthv96** [comment]
```javascript
console.log(
        `Multiline \n string\
         Multiline string\
         Multiline string`
      );
```
Will have the `\n` preserved (As they are escaped in the text representation). This only changes the actual line endings in the file.

**Comment by sidharthv96** [comment]
@thorn0 any pointers on how to handle the codecov failure?

**Comment by thorn0** [comment]
@sidharthv96 Failing because of "+-0.01%" looks like a misconfiguration. Ignore it.

**Comment by thorn0** [comment]
@j-f1 I think your question was about **non-escaped** line breaks in template literals. So I did a little research. Turns out (see the [spec](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-template-literal-lexical-components)) template literals treat any _LineTerminatorSequence_ as LF, no matter how it's represented in the source code: 

image

This means it's totally safe to convert line endings in source files. Prettier's correctness goal isn't compromised.

File: tests/end_of_line/multiline.js (hunk 1 above)
Comment (thorn0) [reply tests/end_of_line/multiline.js]: The change log entry still contains the old snippet though.

**Comment by thorn0** [comment]
@sidharthv96 Thanks for fixing this!

**Pull request merged**